crates/python/src/qpu/result_data.rs

```diff
@@ -43,4 +43,27 @@ impl PyQpuResultData {
     fn readout_values(&self, py: Python<'_>) -> PyResult<HashMap<String, PyReadoutValues>> {
         self.as_inner().readout_values().to_python(py)
     }
+
+    fn asdict(&self, py: Python<'_>) -> PyResult<Py<PyDict>> {
```

Comment:
This makes working with the type on Python much simpler. It is a little bit fragile in that we have to statically name the keys, but the ergonomics are nice and let the user get at an object that works well as a Python type when you don't need/want the extra type safety from the wrapping type.

Reply:
I removed this in favor of a different approach that flattens the object into a struct that can be typed.